### PR TITLE
Impl some of sql connection

### PR DESCRIPTION
**Pull request by EmadAlblueshi**

### CODE REVIEW

File: src/main/java/com/julienviet/pgclient/impl/PreparedTxUpdateCommand.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  private TransactionIsolation isolation;
+
+  PreparedTxUpdateCommand(TransactionIsolation isolation, Handler<AsyncResult<Void>> handler) {
+    this.isolation = isolation;
+    this.handler = handler;
+  }
+
+  @Override
+  boolean exec(DbConnection conn) {
+    switch (isolation) {
```
Comment (vietj) [inline src/main/java/com/julienviet/pgclient/impl/PreparedTxUpdateCommand.java]: user EnumMap instead

Comment (vietj) [inline src/main/java/com/julienviet/pgclient/impl/PreparedTxUpdateCommand.java]: EnumMap<TxIsolation, String>

File: src/main/java/com/julienviet/pgclient/impl/VoidCommandBase.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+import com.julienviet.pgclient.codec.Message;
+import com.julienviet.pgclient.codec.decoder.message.CommandComplete;
+import com.julienviet.pgclient.codec.decoder.message.ErrorResponse;
+
+/**
+ * @author <a href="mailto:emad.albloushi@gmail.com">Emad Alblueshi</a>
+ */
+
+abstract class VoidCommandBase extends CommandBase {
```
Comment (vietj) [inline src/main/java/com/julienviet/pgclient/impl/VoidCommandBase.java]: this class seems useless, please remove it

Comment (EmadAlblueshi) [inline src/main/java/com/julienviet/pgclient/impl/VoidCommandBase.java]: We may need that for BEGIN, COMMIT and ROLLBACK

File: src/main/java/com/julienviet/pgclient/impl/PostgresSQLConnection.java [hunk 3]
```diff
@@ -84,12 +101,12 @@ public void close() {
 
   @Override
   public SQLConnection commit(Handler<AsyncResult<Void>> handler) {
-    throw new UnsupportedOperationException();
+    return this;
```
Comment (vietj) [inline src/main/java/com/julienviet/pgclient/impl/PostgresSQLConnection.java]: I think we should keep the UnsupportedOperationException to mark it's not implemented

File: src/main/java/com/julienviet/pgclient/impl/PostgresSQLConnection.java [hunk 4]
```diff
@@ -84,12 +101,12 @@ public void close() {
...
   }
 
   @Override
   public SQLConnection rollback(Handler<AsyncResult<Void>> handler) {
-    throw new UnsupportedOperationException();
+    return this;
```
Comment (vietj) [inline src/main/java/com/julienviet/pgclient/impl/PostgresSQLConnection.java]: I think we should keep the UnsupportedOperationException to mark it's not implemented

**Comment by EmadAlblueshi** [comment]
👍

**Pull request merged**